pretyping/evarconv.ml

```diff
@@ -1315,12 +1315,17 @@ let thin_evars env sigma sign c =
   let rec applyrec (env,acc) t =
     match kind sigma t with
```

Comment:
The code was wrong and the fix is the wrong fix imo.
The function keeps a state in evdref, but then inspect t using the initial value of the state (sigma).
Indeed `kind` is supposed to always apply the substitution and return `Evar` only when the evar is undefined, so the if-then-else you added is just working around the fact that `kind` is called with the wrong state.

@vgbl cold you pelase try to see if the fix I propose is actually fixing the bug?

Reply:
Thanks for your comment. I’ll try your suggestion.

Reply:
Unfortunately, this does not fix the issue (same anomalies as before)…

Reply:
Then I suspect your old fix was just masking the problem (or I'm completely wrong about what apply rec is supposed to do...)

Reply:
I don't have time now to trace it myself, but I imagine the stack trace is now different

Reply:
Thanks. Here is a second proposal. It only changes `sigma` into `!evdref` in two places. Notice that there are a few occurrences of `sigma` in this code: does someone who understand what’s going on knows whether it is correct to refer to the initial evarmap or not?

Reply:
I think it is never correct... @ppederot

Reply:
Even when raising the `TypingFailed sigma` exception?